cuqipy_fenics/geometry.py

```diff
@@ -398,3 +403,222 @@ def _build_basis(self):
         # Normalize the eigenvectors if required
         if self.normalize:
             self._eig_vec /= np.linalg.norm( self._eig_vec )
+
+
+# Helper user expression to define the step expansion basis
+class _StepExpression(dl.UserExpression):
```

Comment:
It seems straightforward to make it work also under 3D (in the future), right?

Reply:
Yes, I believe this is true, we will need to account for the 3D case in some parts of the code but I do not think it is difficult.. For now, since we have not worked with 3D cases in CUQIpy-FEniCS yet, it is probably something we can address when we need it.